packages/astro/src/runtime/server/endpoint.ts

```diff
@@ -1,28 +1,56 @@
 import type { APIContext, EndpointHandler, Params } from '../../@types/astro';
+import { type LogOptions, warn, info } from '../../core/logger/core.js';
 
-function getHandlerFromModule(mod: EndpointHandler, method: string) {
+function getHandlerFromModule(mod: EndpointHandler, method: string, logging: LogOptions) {
+	const lowerCaseMethod = method.toLowerCase();
+
+	// NOTE: remove in Astro 4.0
+	if (mod[lowerCaseMethod]) {
+		warn(
+			logging,
+			'astro',
+			"Lower case endpoint points are deprecated. In Astro 4.0 it won't work anymore."
+		);
+		info(logging, 'astro', `Rename the endpoint ${lowerCaseMethod} to ${method}`);
```

Comment:
Maybe we can combine the two messages into one?
```suggestion
		warn(
			logging,
			'astro',
			`Lower case endpoint names are deprecated and will not be supported in Astro 4.0. Rename the endpoint ${lowerCaseMethod} to ${method}.`
		);
```